renderdoc/serialise/codecs/vk_cpp_codec/ext_object.h

```diff
@@ -0,0 +1,190 @@
+/******************************************************************************
+* The MIT License (MIT)
+*
+* Copyright (c) 2018 Baldur Karlsson
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+******************************************************************************/
+#pragma once
+
+#include <assert.h>
+#include <math.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <array>
+#include <list>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "common/common.h"
+#include "core/core.h"
+#include "core/intervals.h"
+#include "serialise/rdcfile.h"
+
+namespace vk_cpp_codec
+{
+// Create a fake resource object.
+inline SDObject *makeResourceObject(const char *n, const char *t, uint64_t value)
+{
+  SDObject *ptr = new SDObject(n, t);
+  ptr->type.basetype = SDBasic::Resource;
+  ptr->data.basic.u = value;
+  return ptr;
+}
+
+inline const char *Type(SDObject *ptr)
+{
+  // TODO(akharlamov) Evaluate moving this to filtering stage in TraceTracker class.
+  // This isn't trivial. Patching type.name, when done in Type() is only applied
+  // for SDObjects that store data structures, and patching them during filtering
+  // stage means the filter stage would need to touch on most of Vulkan API OR
+  // crawl through the entire SDObject list and patch every object ignoring Vulkan
+  // specifics.
+  static std::string str = "const char* ";
+  static std::string pipeline_stage_flags = "VkPipelineStageFlags";
+
+  // Just-in-time replacement for types
+  // Vulkan doesn't use std::string, so need to cast it to const char *
+  if(ptr->IsString() || ptr->type.name == "string")
+    return str.c_str();
```

Comment:
Since this function just returns a `const char *` it would be better to just return the literal string directly here, rather than constructing a static `std::string` only to return `c_str()`.

Reply:
Done